.github/workflows/docs-preview-deploy.yml

```diff
@@ -1,100 +1,45 @@
-name: 'Documentation (run)'
+name: 'Docs Preview (Deploy)'
 
 on:
-  # This workflow runs off the primary branch which provides access to the `secrets` context:
-  workflow_run:
-    workflows: ['Documentation (PR)']
-    types:
-      - completed
+  workflow_call:
+    inputs:
+      preview-context:
+        description: 'Preview Metadata (JSON)'
+        required: true
+        type: string
+    secrets:
+      NETLIFY_AUTH_TOKEN:
+        required: true
+      NETLIFY_SITE_ID:
+        required: true
+
+env:
+  BUILD_DIR:  ${{ fromJSON( inputs.preview-context ).build_dir }}
+  # PR head SHA (latest commit):
+  PR_HEADSHA: ${{ fromJSON( inputs.preview-context ).pull_request.head_sha }}
+  PR_NUMBER:  ${{ fromJSON( inputs.preview-context ).pull_request.number }}
+  # Deploy URL preview prefix (the site name for the prefix is managed at Netlify):
+  PREVIEW_SITE_PREFIX: ${{ fromJSON( inputs.preview-context ).netlify.deploy_prefix }}
```

Comment:
Both workflows should be fairly easy to grok now with `inputs` / `env` hoisted up to the top of the workflow.

Let me know if you disagree with the JSON input approach. I could make each one a separate input instead, but that seemed redundant and would just add more noise?